File: backend/ecoWater/ecoWater/src/main/java/com/project/ecoWater/device/domain/Device.java
package com.project.ecoWater.device.domain;


import com.project.ecoWater.tank.app.TankDTO;
import com.project.ecoWater.tank.domain.Tank;
import com.project.ecoWater.tank.infrastructure.TankEntity;
import com.project.ecoWater.user.app.UserDTO;
import com.project.ecoWater.user.domain.User;
import jakarta.persistence.*;
import lombok.*;

import java.sql.Timestamp;
import java.util.UUID;

@Data
@Builder
public class Device {


    private String deviceId;

    private UserDTO user;

    private TankDTO tank;

    private String deviceName;

    private DeviceType deviceType;

    private String deviceLocation;

    private Timestamp creationRegister;

    private Boolean connected;

    private Timestamp lastSeen;

    private String ssid;
}

File: frontend/Proyecto/Proyecto/app/src/main/java/com/example/proyecto/model/device/Device.kt
package com.example.proyecto.model.device

import com.example.proyecto.model.tank.Tank
import com.example.proyecto.model.tank.TankDTO
import com.example.proyecto.model.user.UserDTO

data class Device(
    val deviceId: String?=null,
    val user: UserDTO?=null,
    val deviceName: String?=null,
    val deviceType: DeviceType?=null,
    val deviceLocation: String?=null,
    val creationRegister: String?=null,
    val connected: Boolean?=null,
    val lastSeen: String?=null,
    val ssid: String?=null,
    val tank: Tank?=null
)
